source/communication_interfaces/include/communication_interfaces/sockets/ZMQSocket.hpp

```diff
@@ -32,38 +47,24 @@ class ZMQSocket : public ISocket {
   bool send_bytes(const ByteArray& buffer) override;
 
 protected:
-   /**
-    * @brief Add the parameters with no or default value to the map
-    * @param context The ZMQ context to be used for the sockets
-    * (it's recommended to only use one context per application)
-    */
-  explicit ZMQSocket(const std::shared_ptr<zmq::context_t>& context);
-
+    /**
+     * @brief Constructor taking the configuration struct
+     */
   /**
-   * @brief Add and set the parameters with the parameters given as argument
-   * @param parameters The list of parameters
+   * @brief Constructor taking the configuration struct
+   * @param The configuration struct
    * @param context The ZMQ context to be used for the sockets
-    * (it's recommended to only use one context per application)
+   * (it's recommended to only use one context per application)
    */
-  ZMQSocket(
-      const state_representation::ParameterInterfaceList& parameters, const std::shared_ptr<zmq::context_t>& context
-  );
+  ZMQSocket(ZMQSocketConfiguration configuration, const std::shared_ptr<zmq::context_t>& context);
```

Comment:
The context should be part of the configuration struct if possible, otherwise factory patterns will be harder.

Reply:
Yes that makes sense :+1: